packages/builder/src/builtin-tasks/compile.ts

```diff
@@ -0,0 +1,72 @@
+import { task } from "../internal/core/config/config-env";
+import { networks } from "../../sample-project/algob.config";
+import { TASK_COMPILE } from "./task-names";
+import YAML from "yaml";
+const algosdk = require('algosdk');
+const fs = require('fs');
+const path = require('path');
+const murmurhash = require('murmurhash');
+
+/* Credentials */
+const token = networks.localhost.token;
+const host = networks.localhost.host;
+const port = networks.localhost.port;
+
+let algodclient = new algosdk.Algodv2(token, host, port);
+
+export default function (): void {
+  task(TASK_COMPILE, "Compilation task")
+    .setAction(
+      async () => {
+        console.log("compiling......");
+        /*File paths for assets/ and artifacts/cache/ */
```

Comment:
Do not add these comments. Instead create multiple methods and their names will be actual documentation.
These comments don't add value:

// I do this
this
// I do that
that
// I do the rest
1
2
// some random file IO
echo hi > output.txt
3
4
// Hi mom
mom